### PR TITLE
Rename family sample

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR renames the FamilySample ~table~ model to CaseSample and any references to it.

### Fixed
- Rename FamilySample
- Bonus: rename family parameter on relate_sample

### How to test

- [x] Do ensure all the views work as expected

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/store/models.py [hunk 1]
```diff
@@ -515,7 +515,7 @@ def to_dict(self, links: bool = False, analyses: bool = False) -> dict:
         return data
 
 
-class FamilySample(Model):
+class CaseSample(Model):
     __tablename__ = "family_sample"
```
Comment (ChrOertlin) [inline cg/store/models.py:519]:
```suggestion
    __tablename__ = "case_sample"
```

File: cg/store/models.py [hunk 2]
```diff
@@ -515,7 +515,7 @@ def to_dict(self, links: bool = False, analyses: bool = False) -> dict:
...
     __table_args__ = (UniqueConstraint("family_id", "sample_id", name="_family_sample_uc"),)
```
Comment (ChrOertlin) [inline cg/store/models.py:520]:
```suggestion
    __table_args__ = (UniqueConstraint("case_id", "sample_id", name="_case_sample_uc"),)
```

**Review by ChrOertlin — APPROVED** [review]
Looks good! Just a thing about database table names, if not required yet you can ignore them!

File: cg/store/models.py (hunk 2 above)
Comment (seallard) [reply cg/store/models.py:520]: Not remaining the table stuff yet, my PR description was a bit off

**Pull request merged**